src/solvers/nonlinear_solid_solver.hpp

```diff
@@ -73,7 +69,7 @@ class NonlinearSolidSolver : public BaseSolver {
   void SetTractionBCs(std::vector<int> &trac_bdr, mfem::VectorCoefficient *trac_bdr_coef);
 
   /// Set the viscosity coefficient
-  void SetViscosity(mfem::Coefficient *visc_coef);
+  void SetViscosity(std::shared_ptr<mfem::Coefficient> visc_coef);
```

Comment:
@white238 @samuelpmish @jonwong12 Any thoughts on how to not require a user to use `std::shared_ptr`? I would like to have a pointer-type object of the base `mfem::Coefficient` class, but I can't figure out how to get `std::shared_ptr` to work with an argument of type `mfem::Coefficient &`.

Reply:
To summarize our webex: we decided making the user pass a `shared_ptr` is the way to go.